R/example.R

```diff
@@ -0,0 +1,75 @@
+# Kaiaulu - https://github.com/sailuh/kaiaulu
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+# This file is meant to generate fake git data
+
+
+#'  A repo with 3 commits. The first adds hello.R.
+#' The second renames the file to hi.R. The third adds a second file bye.R
+#'
+#' @return git_repo of newly created empty repo
+#' @export
+example_repo_three_commits <- function() {
+
+# Create folder & repo
+folder_path <- io_create_folder("example_repo")
+git_init(folder_path)
+git_repo <- file.path(folder_path, '.git')
+
+# Add hello.R file and commit it
+hello_path <- file.path(folder_path, "hello.R")
+io_make_sample_file(hello_path, "print('hello!')")
+git_add(git_repo, folder_path, hello_path)
+git_commit(git_repo, folder_path, "Commit hello.R file to empty repo", "John Doe", "JohnDoe@test.com")
+
+
+# rename the file from hello.R to hi.R, then add and commit
+git_rename("hello.R", "hi.R")
+hi_path <- file.path(folder_path, "hi.R")
+git_add(git_repo, folder_path, hi_path)
+git_commit(git_repo, folder_path, "Renamed file name to hi.R", "John Doe", "JohnDoe@test.com")
+
+# Add bye.R file and commit it
+bye_path <- file.path(folder_path, "bye.R")
+io_make_sample_file(bye_path, "print('bye!')")
+git_add(git_repo, folder_path, bye_path)
+git_commit(git_repo, folder_path, "Commit bye.R file to repo", "John Doe", "JohnDoe@test.com")
+
+}
+
+
+# A repo with 3 commits, where 1 file has as prefix _test.R, 1 file has the suffix example_*.R,
+# and 1 file hello.R.
+example_repo_suffix_prefix <- function() {
+
+  # Create folder & repo
+  folder_path <- io_create_folder("example_repo")
+  git_init(folder_path)
+  git_repo <- file.path(folder_path, '.git')
+
+  # Add example_test.R file and commit it
+  test_path <- file.path(folder_path, "test-example.R")
+  io_make_sample_file(test_path, "print('tester')")
+  git_add(git_repo, folder_path, test_path)
+  git_commit(git_repo, folder_path, "Commit test-example.R file to repo", "John Doe", "JohnDoe@test.com")
+
+
+  # Add hello.R file and commit it
+  example_path <- file.path(folder_path, "fake-example.R")
```

Comment:
This is not the intended file format. The idea was to create two files with different prefixes, test, and example, and try to filter, resulting in the expected equal 1. Failing the test would lead to 3 files.

Reply:
Sorry for the wrong file naming conventions, I misunderstood the issue description. So just to clarify, both "example" and "test" are prefixes for files. Neither of them are suffixes? So, some test files I could create to test my filter functions include "test-fakedata.R" and "example-fakedata.R"?

Reply:
the suffix is the file extension, e.g. `.R` or `.java` or `.txt`. Have a look at the docs of the filter functions. In a real analysis, you don't want to be reporting the number of enhancements the `README.txt` has, but only the source code. The same is true for unit test or example files. Seldom they are the primarily object of interest when analyzing the quality of the code (unless you are explicitly interest on the unit tests quality). 

So the goal of this example was to create the 3 files, and use 2 filters. In the commit I did for you, the 2 prefix filters are utilized and correctly filtered. The suffix filter is not implemented yet, but to be honest, that is fine. I'd say, just polish the name of the variables and the docs on the example function to be a test of prefixes instead of suffix! 

There is something on this unit test I will need to change later (not due to a mistake on your part) anyway, so I'd prefer you just to make sure you understand what is going on the code, and that every line the example and unit test executes you check on the terminal it is working as intended as-is.